netbox/dcim/api/views.py

```diff
@@ -200,6 +203,39 @@ def units(self, request, pk=None):
             rack_units = serializers.RackUnitSerializer(page, many=True, context={'request': request})
             return self.get_paginated_response(rack_units.data)
 
+    @swagger_auto_schema(
+        responses={200: serializers.RackUnitSerializer(many=True)},
+        query_serializer=serializers.RackElevationDetailFilterSerializer
+    )
+    @action(detail=True)
+    def elevation(self, request, pk=None):
+        """
+        Rack elevation representing the list of rack units. Also supports rendering the elevation as an SVG.
+        """
+        rack = get_object_or_404(Rack, pk=pk)
+        serializer = serializers.RackElevationDetailFilterSerializer(data=request.GET)
+        if not serializer.is_valid():
+            return Response(serializer.errors, 400)
+        data = serializer.validated_data
+
+        if data['render_format'] == 'svg':
+            drawing = rack.get_elevation_svg(data['face'], data['width'], data['unit_height'])
+            return HttpResponse(drawing.tostring(), content_type='image/svg+xml')
+
+        else:
+            elevation = rack.get_rack_units(
+                face=data['face'],
+                exclude=data['exclude'],
+                expand_devices=data['expand_devices']
+            )
+            if data['q']:
+                elevation = [u for u in elevation if data['q'] in str(u['id'])]
```

Comment:
I don't understand the purpose of this. It seems to be matching `q` against the numeric unit number, which doesn't seem useful. I'm not sure why we would ever return a partial elevation anyway. I could be misunderstanding.

Reply:
As @lampwins pointed out, this behavior is present in the current `units` endpoint, apparently for filtering by unit ID. I don't believe this functionality is used any longer.